src/util-logopenfile.c

```diff
@@ -828,10 +833,12 @@ static bool LogFileNewThreadedCtx(LogFileCtx *parent_ctx, const char *log_path,
         thread->Write = SCLogFileWriteNoLock;
         thread->Close = SCLogFileCloseNoLock;
         OutputRegisterFileRotationFlag(&thread->rotation_flag);
-    } else if (parent_ctx->type == LOGFILE_TYPE_PLUGIN) {
+    } else if (parent_ctx->type == LOGFILE_TYPE_FILETYPE) {
         entry->slot_number = SC_ATOMIC_ADD(eve_file_id, 1);
-        thread->plugin.plugin->ThreadInit(
-                thread->plugin.init_data, entry->slot_number, &thread->plugin.thread_data);
+        SCLogDebug("%s - thread %d [slot %d]", log_path, entry->internal_thread_id,
+                entry->slot_number);
+        thread->filetype.filetype->ThreadInit(thread->filetype.init_data, entry->internal_thread_id,
+                &thread->filetype.thread_data);
```

Comment:
@jlucovsky Can you review this change around the threaded stuff..  So the filetype no longer has to worry about it - other than being thread safe of course.

Reply:
Will do. The plugin/filetype should never be concerned with threading outside of its own implementation-specific threading.

Reply:
> Will do. The plugin/filetype should never be concerned with threading outside of its own implementation-specific threading.

without this pr they all have to, with this pr a plugin that is written for threaded mode will also run in non-threaded mode which isn't true in current git master.